### PR TITLE
(PC-6698) add animation select date

### PR DESCRIPTION
**Pull request by emmajmn**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-6698

## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [ ] Written **unit tests** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [x] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] |
| -------------------- | 
| [gif](https://user-images.githubusercontent.com/50346870/110126425-b34a8700-7dc4-11eb-95c3-384639dd6c79.mov)|                          | 

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: src/features/bookOffer/components/Calendar/Calendar.tsx [hunk 1]
```diff
@@ -76,6 +88,13 @@ interface Props {
 export const Calendar: React.FC<Props> = ({ stocks, userRemainingCredit }) => {
   const { bookingState, dispatch } = useBooking()
   const stocksDate = getStocksByDate(stocks)
+  const debouncedDispatch = useRef(debounce(dispatch, 1000)).current
```
Comment (antoinewg) [inline src/features/bookOffer/components/Calendar/Calendar.tsx:91]:
```suggestion
  const debouncedDispatch = useRef(debounce(dispatch, 200)).current
```

sinon on dirait que c'est un bug

File: src/features/bookOffer/components/Calendar/Calendar.tsx [hunk 2]
```diff
@@ -25,12 +26,25 @@ LocaleConfig.locales['fr'] = {
 }
 LocaleConfig.defaultLocale = 'fr'
 
+export const formatToKeyDate = (ISODate: string) => {
```
Comment (antoinewg) [inline src/features/bookOffer/components/Calendar/Calendar.tsx:29]:
```suggestion
export const formatToKeyDate = (date: Date) => {
```

comme ça tu peux supprimer `const date = new Date(ISODate)`.

Vois si tu peux pas déplacer cette fonction dans un fichier plus adapté.

File: src/features/bookOffer/components/Calendar/Calendar.tsx [hunk 3]
```diff
@@ -76,6 +88,13 @@ interface Props {
 export const Calendar: React.FC<Props> = ({ stocks, userRemainingCredit }) => {
   const { bookingState, dispatch } = useBooking()
   const stocksDate = getStocksByDate(stocks)
+  const debouncedDispatch = useRef(debounce(dispatch, 1000)).current
+
+  const markedDates: { [keyDate: string]: { selected: boolean; marked: boolean } } = {}
+  if (bookingState.date) {
+    const keyDate = formatToKeyDate(bookingState.date.toString())
+    markedDates[keyDate] = { selected: true, marked: true }
```
Comment (antoinewg) [inline src/features/bookOffer/components/Calendar/Calendar.tsx:96]: je sais pas si marked sert à qqc

Comment (emmajmn) [reply src/features/bookOffer/components/Calendar/Calendar.tsx:96]: hum je sais pas trop .. je vais essayer en le supprimant pour voir

File: src/features/bookOffer/components/Calendar/Calendar.tsx (hunk 2 above)
Comment (emmajmn) [reply src/features/bookOffer/components/Calendar/Calendar.tsx:29]: ok je modifie

**Review by antoinewg — APPROVED** [review]

**Pull request merged**